runtime-watcher/internal/handler.go

```diff
@@ -259,25 +261,44 @@ func (h *Handler) sendRequestToKcp(moduleName string, watched WatchedObject) err
 	if err != nil {
 		return h.logAndReturnKCPErr(err, watchermetrics.ReasonRequest)
 	}
-	resp, err := httpsClient.Post(url, "application/json", bytes.NewBuffer(postBody))
-	if err != nil {
-		err = errors.Join(errKcpRequest, err)
-		h.logger.Error(err, err.Error(), "postBody", watcherEvent)
-		h.metrics.UpdateFailedKCPTotal(watchermetrics.ReasonResponse)
-		return err
+
+	maxAttempts := 3
+	for currentAttempt := 1; currentAttempt <= maxAttempts; currentAttempt++ {
```

Comment:
* We should actually only retry the http call, not reading the body and not on statuscode.

* Can you check if the library https://github.com/sethgrid/pester is a usable to retry the calls? (they have a `func NewExtendedClient(hc *http.Client)`)

* And let's add a bit of backoff to the retry. 
If you do and the previously mentioned lib is not usable,
another alternative is using https://github.com/avast/retry-go then and use the delay functionality there instead of writing it ourselves:
```
retryOpts := []retry.Option{
	retry.Attempts(3),
	retry.DelayType(retry.BackOffDelay),
}

err := retry.Do(func() error {
	resp, err := httpsClient.Post( ... )
        ...
}, retryOpts...)
```

Reply:
pester is a cool library but I wanted to avoid more dependencies. However, if you see it fit then I will use it with backoff